pkg/validation/limits.go

```diff
@@ -173,6 +173,8 @@ type Limits struct {
 
 	RequiredLabels       []string `yaml:"required_labels,omitempty" json:"required_labels,omitempty" doc:"description=Define a list of required selector labels."`
 	RequiredNumberLabels int      `yaml:"minimum_labels_number,omitempty" json:"minimum_labels_number,omitempty" doc:"description=Minimum number of label matchers a query should contain."`
+
+	IndexGatewayShardSize int `yaml:"gateway_shard_size" json:"gateway_shard_size"`
```

Comment:
ditto: to be more consistent, maybe you can use "index_gateway_shard_size" or change the cli flag from "index-gateway.shard-size" to "gateway.shard-size".